packages/main/src/plugin/proxy.ts

```diff
@@ -22,6 +22,13 @@ import { ProxyAgent } from 'undici';
 import type { ConfigurationRegistry, IConfigurationNode } from './configuration-registry.js';
 import { Emitter } from './events/emitter.js';
 import { getProxyUrl } from './proxy-resolver.js';
+import { getProxySettingsFromSystem } from './proxy-system.js';
+
+export enum ProxyState {
```

Comment:
If we need this to be shared between renderer and main, we might place it in a dedicated file in `packages/api/src`, therefore both renderer and main could access it using /@api/...